src/services/solana-client/createUserBank.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Account,
+  AccountMeta,
+  Connection,
+  PublicKey,
+  SystemProgram,
+  SYSVAR_RENT_PUBKEY
+} from '@solana/web3.js'
+import bs58 from 'bs58'
+import BN from 'bn.js'
+import {
+  solanaClusterEndpoint,
+  mintKey,
+  tokenProgramKey,
+  audiusProgramPubkey,
+  generatedProgramPDA
+} from './config'
+import SolanaClient from './index'
+
+export const getBankAccountAddress = async (ethAddress: string) => {
+  const baseAccount = generatedProgramPDA
+
+  // Solana client likes eth addresses as an array of uints
+  const rawEthAddress = new BN(ethAddress, 'hex')
+
+  // We b58 encode our eth address to use as seed later on
+  const b58EthAddress = bs58.encode(rawEthAddress.toArrayLike(Buffer))
+
+  const accountToGenerate: PublicKey = await PublicKey.createWithSeed(
+    /* from pubkey / base */ baseAccount,
+    /* seed */ b58EthAddress,
+    /* programId / owner */ tokenProgramKey
+  )
+  return accountToGenerate
+}
+const feePayerPrivateKey = Uint8Array.from(
+  JSON.parse(process.env.REACT_APP_PRIVATE || '')
+)
+console.log({ feePayerPrivateKey })
+// createUserBank deterministically creates a Solana wAudio token account
+// from an ethAddress (without the '0x' prefix)
+export async function createUserBankFrom(ethAddress: string) {
+  // Solana client likes eth addresses as an array of uints
+  const rawEthAddress = new BN(ethAddress, 'hex')
+  console.log({ rawEthAddress })
+
+  // We need to prepend a zero hero so rust knows which enum case we're dealing with
+  // https://paulx.dev/blog/2021/01/14/programming-on-solana-an-introduction/
+  const ethAddressInstruction = Buffer.from(
+    Uint8Array.of(0, ...rawEthAddress.toArray('be'))
+  )
+  // We b58 encode our eth address to use as seed later on
+  const b58EthAddress = bs58.encode(rawEthAddress.toArrayLike(Buffer))
```

Comment:
since we have our `getBankAccountAddress` function now, don't think we need to b58 encode this outside anymore